### PR TITLE
[cassandra] Fix batched query leak

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
The cassandra integration does not adhere spans created for batch statements to the specification required by the agent to obfuscate the query.

This PR changes the logic so that query is set as the resource of the span. Since the span type is already set to `'cassandra'` the query can now be properly obfuscated by the agent.

Before:
screen shot 2018-11-15 at 3 07 02 pm

After:
screen shot 2018-11-15 at 2 47 20 pm

### CODE REVIEW

File: tests/contrib/cassandra/test.py [hunk 1]
```diff
@@ -264,7 +264,11 @@ def test_batch_statement(self):
         spans = writer.pop()
         eq_(len(spans), 1)
         s = spans[0]
-        eq_(s.resource, 'BatchStatement')
+        eq_(
+            s.resource,
+            """INSERT INTO test.person_write (name, age, description) VALUES ('Joe', 1, 'a'); """
```
Comment (brettlangdon) [inline tests/contrib/cassandra/test.py]: I like to put parens around multiline strings, helps to reduce confusion on whether these are separate or joined.

```python

```
eq_(
    s.resource,
    (
         'INSERT INTO test.person_write (name, age, description) VALUES (\'Joe'\, 1, \'a\'); '
         'INSERT INTO test.person_write (name, age, description) VALUES (\'Jane\', 2, \'b\')'
    ),
)

**Comment by Kyle-Verhoog** [comment]
Yes this was tested with the agent obfuscation. However your comment made me check again and I caught an issue where including the query as a tag for a cassandra span will not result in the query being overwritten with the obfuscated query.

I've updated the PR description with the before and after of a trace.

One more thing to consider is if there are issues with using multiple combined queries as a resource.

File: tests/contrib/cassandra/test.py [hunk 2]
```diff
@@ -264,9 +264,15 @@ def test_batch_statement(self):
...
+        eq_(
+            s.resource,
+            (
+                'INSERT INTO test.person_write (name, age, description) VALUES (\'Joe\', 1, \'a\'); '
+                'INSERT INTO test.person_write (name, age, description) VALUES (\'Jane\', 2, \'b\')'
+            )
+        )
         eq_(s.get_metric('cassandra.batch_size'), 2)
-        assert 'test.person' in s.get_tag('cassandra.query')
+        assert 'test.person' in s.resource
```
Comment (brettlangdon) [inline tests/contrib/cassandra/test.py]: we should probably be asserting that this shows up twice in the resource?

Comment (brettlangdon) [reply tests/contrib/cassandra/test.py]: or, we should know what the exact resource is, no? or does it change between versions?

**Comment by brettlangdon** [comment]
Also, have you tested with running multiple queries at once in batch?

Either 2-3 of the same query or 2-3 of different queries. do we handle obfuscation properly there?

Other than that and the one question I had re: assertion, this appears to be mostly good to go.

File: tests/contrib/cassandra/test.py (hunk 2 above)
Comment (Kyle-Verhoog) [reply tests/contrib/cassandra/test.py]: This can actually be removed since we test the resource exactly on line 267.

**Comment by Kyle-Verhoog** [comment]
Obfuscation seems to work with multiple queries, including different types of queries.

File: tests/contrib/cassandra/test.py [hunk 3]
```diff
@@ -264,9 +264,14 @@ def test_batch_statement(self):
...
-        eq_(s.resource, 'BatchStatement')
+        eq_(
+            s.resource,
+            (
+                'INSERT INTO test.person_write (name, age, description) VALUES (\'Joe\', 1, \'a\'); '
+                'INSERT INTO test.person_write (name, age, description) VALUES (\'Jane\', 2, \'b\')'
+            )
+        )
         eq_(s.get_metric('cassandra.batch_size'), 2)
-        assert 'test.person' in s.get_tag('cassandra.query')
```
Comment (brettlangdon) [inline tests/contrib/cassandra/test.py]: actually, should this be `assert s.get_tag('cassandra.query') is None`? to ensure we aren't setting this tag anymore?

**Comment by brettlangdon** [comment]
One last small question, but otherwise this is good to go.

**Comment by brettlangdon** [comment]
There is a merge conflict here now as well.

File: ddtrace/contrib/cassandra/session.py [hunk 4]
```diff
@@ -253,7 +253,6 @@ def _sanitize_query(span, query):
         # reset query if a string is available
         resource = getattr(query, "query_string", query)
     elif t == 'BatchStatement':
-        resource = 'BatchStatement'
```
Comment (brettlangdon) [inline ddtrace/contrib/cassandra/session.py]: bad merge?

**Pull request merged**